src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -183,7 +184,11 @@ function ($array) {
     protected function assignSortFields($queryBuilder)
     {
         foreach ($this->sortFields as $index => $sortField) {
-            $queryBuilder->addOrderBy($sortField->getSelect(), $this->sortOrders[$index]);
+            $sort = ($sortField instanceof DoctrineConcatenationFieldDescriptor)
+                ? $sortField->getName()
+                : $sortField->getSelect();
```

Comment:
@danrot That's how I approached this originally but it appeared to break a bunch of other bundles (https://travis-ci.org/sulu-io/sulu/jobs/77311121), I didn't have the time to investigate it further at that point so compromised with the ternary.